app/javascript/mastodon/hooks/useAccountId.ts

```diff
@@ -33,5 +32,9 @@ export function useAccountId() {
     }
   }, [dispatch, accountId, acct, isAccount]);
 
-  return accountId;
+  if (accountId) {
+    return accountId;
+  }
+
+  return undefined;
```

Comment:
Maybe the following might be a little easier to read?
```suggestion
  return accountId ?? undefined;
```

Reply:
But overall, the previous version had an actual issue where it could return a string, `undefined` or `null` but the types were overridden so Typescript was considering only `string` and `undefined`. I think in some places (probably non-Typescript only) we rely on the difference between `undefined` and `null`, as `undefined` means “we haven't fetched it yet” and `null` means “it does not exist”.

If I'm wrong and we never make the distinction, we probably should never store `null` and instead delete from the map, simplifying the type and having `string | undefined` everywhere.

Reply:
We indeed do make a distinction in at least `app/javascript/mastodon/features/account_timeline/index.jsx`, with `https://mastodon.social/@bogus@example.com` displaying a 404, but `app/javascript/mastodon/features/account_featured/index.tsx`, not making that distinction, displays an infinite spinner: `https://mastodon.social/@bogus@example.com/featured`

Reply:
Thanks, I think I made it better!